cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -764,11 +844,79 @@ private void createBanners(Collection<Banner> banners) {
             break;
           }
         }
+      }
+
+      boolean targetEvents = false;
+      if (banner.getEventFilters() != null && banner.getEventFilters().size() > 0) {
+        isTargetEvent = true;
+        for (BannerTargetEvent bannerTargetEvent : banner.getEventFilters()) {
+          boolean targetConditionTrue;
+          String event = bannerTargetEvent.getEvent();
+          String property = bannerTargetEvent.getProperty();
+          String relationString = bannerTargetEvent.getRelation();
+          String value = bannerTargetEvent.getValue();
+          String fromValue = bannerTargetEvent.getFromValue();
+          String toValue = bannerTargetEvent.getToValue();
+
+          if (event == null || event.isEmpty() ||
```

Comment:
what is this monster if condition
lets create a function isValidInBetweenNumber/Double dont know you name it
function paramter
return paramter != null && !paramter.isEmpty() && Double.parseDouble(paramter) >= 0 && Double.parseDouble(paramter) % 1 == 0
not sure dought why it has to be >= 0, also add comment about this function
// currently we only accept whole numbers here since the amount of how often an event can be triggered can not be a floating number